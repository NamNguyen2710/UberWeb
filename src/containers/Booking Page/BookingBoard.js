import location_marker from '../../images/location-marker.png'

function BookingBoard(props) {
  return (
    <div className="confirm-board">
      <div className="row" onClick={() => {props.revertToSearch("from")}}>
        <img src={location_marker} alt="from" />
        <div className="col">
          <h4>From</h4>
          {props.from.info}
        </div>
      </div>
      <div className="row" onClick={() => {props.revertToSearch("to")}}>
        <img src={location_marker} alt="to"/>
        <div className="col">
          <h4>To</h4>
          {props.to.info}
        </div>
      </div>
      <div className="row">
        <div className="col">
          <h4>Prize: $9,76</h4>
          <p>Kilometer: 3.85km</p>
          <p>Time: 35 minutes</p>
        </div>
        <div>
          Fare Breakdown
        </div>
      </div>
      <div className="btn-row">
        <button className="back-btn" onClick={() => {props.revertToSearch("to")}}>← Back</button>
        <button className="black-btn" onClick={props.confirmBooking}>Confirm →</button>
      </div>
    </div>
  );
}

export default BookingBoard;