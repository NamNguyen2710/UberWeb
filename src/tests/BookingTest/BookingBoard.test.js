import { waitFor } from "@testing-library/react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import BookingBoard from "../../containers/BookingPage/BookingBoard";

describe("Booking board Booking flow", () => {
  it("render correctly", () => {
    const revertToSearch = jest.fn();
    const booking = jest.fn();
    const from = { info: "" };
    const to = { info: "" };
    const snapshot = renderer
      .create(
        <BookingBoard
          from={from}
          to={to}
          revertToSearch={revertToSearch}
          confirmBooking={booking}
        />
      )
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it("revert and submit", async () => {
    const revertToSearch = jest.fn();
    const booking = jest.fn();
    const from = { info: "" };
    const to = { info: "" };
    const wrapper = mount(
      <BookingBoard
        from={from}
        to={to}
        revertToSearch={revertToSearch}
        confirmBooking={booking}
      />
    );

    wrapper.find(".row").at(0).props().onClick();
    await waitFor(() => {
      expect(revertToSearch).toBeCalledWith("from");
    });

    wrapper.find(".row").at(1).props().onClick();
    await waitFor(() => {
      expect(revertToSearch).toBeCalledWith("to");
    });

    wrapper.find(".back-btn").props().onClick();
    await waitFor(() => {
      expect(revertToSearch).toBeCalledWith("to");
    });

    wrapper.find(".black-btn").props().onClick();
    await waitFor(() => {
      expect(booking).toHaveBeenCalled();
    });
  });
});
