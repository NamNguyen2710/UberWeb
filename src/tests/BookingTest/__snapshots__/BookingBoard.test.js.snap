// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking board Booking flow render correctly 1`] = `
<div
  className="confirm-board"
>
  <div
    className="row"
    onClick={[Function]}
  >
    <img
      alt="from"
      src="location-marker.png"
    />
    <div
      className="col"
    >
      <h4>
        From
      </h4>
      
    </div>
  </div>
  <div
    className="row"
    onClick={[Function]}
  >
    <img
      alt="to"
      src="location-marker.png"
    />
    <div
      className="col"
    >
      <h4>
        To
      </h4>
      
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h4>
        Prize: $9,76
      </h4>
      <p>
        Kilometer: 3.85km
      </p>
      <p>
        Time: 35 minutes
      </p>
    </div>
    <div>
      Fare Breakdown
    </div>
  </div>
  <div
    className="btn-row"
  >
    <button
      className="back-btn"
      onClick={[Function]}
    >
      ← Back
    </button>
    <button
      className="black-btn"
      onClick={[MockFunction]}
    >
      Confirm →
    </button>
  </div>
</div>
`;
